### PR TITLE
feat(kms): add integrity verification to samples of crypto operations

### LINKED ISSUE #4284: kms: Add integrity verification to samples crypto operations
Adding guidelines to the samples of Cloud KMS's Java Client Library for performing request and response integrity verification when using cryptographic operations.

Each one of the operation symmetric encrypt, symmetric decrypt, asymmetirc sign, asymmetric decrypt, get public key should do the following:

Populate CRC32C fields in the request object.
Verify integrity fields in the response object.

Context: https://cloud.google.com/kms/docs/data-integrity-guidelines

### PR DESCRIPTION
**Pull request by iamtamjam**
Fixes #4284

> It's a good idea to open an issue first for discussion.

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Review by lesv — APPROVED** [review]

**Comment by iamtamjam** [comment]
@lesv I finalized my updates to KMS's EncryptSymmetric.java sample to add request/response integrity verification. The tests fail occasionally, I think due to quota errors, which manifest themselves as a false fail. 

I want to repeat these additions in other KMS samples before marking the PR as ready for review. Could you please give my updates in `EncryptSymmetric.java` an early look for any style and readability comments? This will help me apply your comments to the other files faster.

Thank you!

**Comment by iamtamjam** [comment]
@shubha-rajan I see that lesv@ is out for the next 2 weeks. Could you please review this PR in his place? 

This PR fails occasionally, which makes me suspect that it hits flaky errors, such as quota. Any advice on how I can get around it? If you could tell me what GCP project is  used to for the KMS tests (it seems to be set by an env var GOOGLE_CLOUD_PROJECT), I can increase this project's quota for KMS operations.

File: kms/src/main/java/kms/EncryptSymmetric.java [hunk 1]
```diff
@@ -38,20 +45,63 @@ public void encryptSymmetric() throws IOException {
...
+          response.getCiphertextCrc32C().getValue(), getCrc32cAsLong(response.getCiphertext().toByteArray()),
+          getCrc32cAsLong("test"));
+      throw new IOException("Encrypt: response from server corrupted. " + details);
+    }
+
+    System.out.printf("Ciphertext: %s%n", response.getCiphertext().toStringUtf8());
+  }
+
+  private long getCrc32cAsLong(byte[] data) {
+    return (long) Hashing.crc32c().hashBytes(data).asInt();
```
Comment (shubha-rajan) [inline kms/src/main/java/kms/EncryptSymmetric.java:100]:
```suggestion
    return (long) Hashing.crc32c().hashBytes(data).padToLong();
```

File: kms/src/main/java/kms/GetPublicKey.java [hunk 2]
```diff
@@ -51,8 +55,30 @@ public void getPublicKey(
...
+          publicKey.getPemBytes().toByteArray())) {
+        throw new IOException("GetPublicKey: response from server corrupted");
+      }
+
       System.out.printf("Public key: %s%n", publicKey.getPem());
     }
   }
+
+  private long getCrc32cAsLong(byte[] data) {
+    return (long) Hashing.crc32c().hashBytes(data).asInt();
```
Comment (shubha-rajan) [inline kms/src/main/java/kms/GetPublicKey.java:77]:
```suggestion
    return (long) Hashing.crc32c().hashBytes(data).padToLong();
```

File: kms/src/main/java/kms/SignAsymmetric.java [hunk 3]
```diff
@@ -70,14 +75,44 @@ public void signAsymmetric(
...
       // Get the signature.
-      byte[] signature = result.getSignature().toByteArray();
+      byte[] signature = response.getSignature().toByteArray();
 
       System.out.printf("Signature %s%n", signature);
     }
   }
+
+  private long getCrc32cAsLong(byte[] data) {
+    return (long) Hashing.crc32c().hashBytes(data).asInt();
```
Comment (shubha-rajan) [inline kms/src/main/java/kms/SignAsymmetric.java:111]:
```suggestion
    return (long) Hashing.crc32c().hashBytes(data).padToLong();
```

File: kms/src/main/java/kms/DecryptSymmetric.java [hunk 4]
```diff
@@ -44,14 +51,40 @@ public void decryptSymmetric(
...
+          response.getPlaintext().toByteArray())) {
+        throw new IOException("Decrypt: response from server corrupted, getCrc32cAsLong(test)=" + getCrc32cAsLong("test"));
+      }
+
       System.out.printf("Plaintext: %s%n", response.getPlaintext().toStringUtf8());
     }
   }
+
+  private long getCrc32cAsLong(byte[] data) {
+    return (long) Hashing.crc32c().hashBytes(data).asInt();
```
Comment (shubha-rajan) [inline kms/src/main/java/kms/DecryptSymmetric.java:83]:
```suggestion
    return (long) Hashing.crc32c().hashBytes(data).padToLong();
```

File: kms/src/main/java/kms/DecryptAsymmetric.java [hunk 5]
```diff
@@ -56,11 +61,41 @@ public void decryptAsymmetric(
...
+          response.getPlaintext().toByteArray())) {
+        throw new IOException("AsymmetricDecrypt: response from server corrupted");
+      }
+
       System.out.printf("Plaintext: %s%n", response.getPlaintext().toStringUtf8());
     }
   }
+
+  private long getCrc32cAsLong(byte[] data) {
+    return (long) Hashing.crc32c().hashBytes(data).asInt();
```
Comment (shubha-rajan) [inline kms/src/main/java/kms/DecryptAsymmetric.java:94]:
```suggestion
    return (long) Hashing.crc32c().hashBytes(data).padToLong();
```

Comment (shubha-rajan) [inline kms/src/main/java/kms/DecryptAsymmetric.java:94]: Any reason you're using `asInt()` here? I tried this with `padToLong()' and the tests started to consistently pass. https://guava.dev/releases/19.0/api/docs/com/google/common/hash/HashCode.html#padToLong()

**Review by shubha-rajan — CHANGES_REQUESTED** [review]

**Comment by iamtamjam** [comment]
@shubha-rajan Your suggestion worked! Thank you!! The PR is ready for review.

File: kms/src/main/java/kms/DecryptAsymmetric.java [hunk 6]
```diff
@@ -56,11 +61,41 @@ public void decryptAsymmetric(
       CryptoKeyVersionName keyVersionName =
           CryptoKeyVersionName.of(projectId, locationId, keyRingId, keyId, keyVersionId);
 
+      // Optional, but recommended: compute ciphertext's CRC32C. See helpers below.
+      long ceiphertextCrc32c = getCrc32cAsLong(ciphertext);
```
Comment (shubha-rajan) [inline kms/src/main/java/kms/DecryptAsymmetric.java:65]: This spelling error shows up in a few other places. I'd suggest doing a global find/replace
```suggestion
      long ciphertextCrc32c = getCrc32cAsLong(ciphertext);
```

**Review by shubha-rajan — CHANGES_REQUESTED** [review]
This is almost ready, just need to fix one spelling mistake in a variable name that pops up a few times.

File: kms/src/main/java/kms/DecryptAsymmetric.java (hunk 6 above)
Comment (iamtamjam) [reply kms/src/main/java/kms/DecryptAsymmetric.java]: Thanks for catching that. Performed a global search/replace.

**Review by shubha-rajan — APPROVED** [review]

**Review by lesv — APPROVED** [review]

**Pull request merged**